clang/lib/StaticAnalyzer/Checkers/StreamChecker.cpp

```diff
@@ -639,12 +644,7 @@ void StreamChecker::preFreadFwrite(const FnDescription *Desc,
                                    bool IsFread) const {
   ProgramStateRef State = C.getState();
   SVal StreamVal = getStreamArg(Desc, Call);
-  State = ensureStreamNonNull(StreamVal, Call.getArgExpr(Desc->StreamArgNo), C,
-                              State);
-  if (!State)
-    return;
-  State = ensureStreamOpened(StreamVal, C, State);
-  if (!State)
+  if (!basicCheck(Desc, Call, C, State, StreamVal))
```

Comment:
```suggestion
  State = ensureStreamNonNullAndOpened(Desc, Call, C, State, StreamVal);
  if (!State)
    return;
```

Reply:
The function returns a state that is modified further by the following functions. Otherwise the state changes applied in `basicCheck` are lost.  (`ensureStreamNonNull` does a state change, `ensureStreamOpened` does not, but all return a state to make the usage similar.)